projects/vdk-plugins/vdk-jupyter/.plugin-ci.yml

```diff
@@ -33,6 +33,41 @@ build-vdk-jupyterlab-extension:
         - "projects/vdk-plugins/vdk-jupyter/**/*"
 
 
+test-vdk-jupyterlab-extension:
+  stage: test
+  image: "python:3.8-alpine"
+  before_script:
+    - cd projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension || exit 1
+    - pip install -U pip setuptools pre-commit
+    - apk --no-cache add npm rust cargo libffi-dev
+    - pip install jupyter
+  script:
+    - echo "Running Server tests..."
+    - pip install -e ".[test]"
+    - pytest -vv -r ap --cov vdk-jupyterlab-extension
+
+    - echo "Running Frontend tests..."
+    - jlpm
+    - jlpm test
+
+    - echo "Running Integration tests..."
+    - jlpm install
+    - jlpm build:prod
+    - cd ./ui-tests
+    - jlpm install
+    - jlpm playwright install
+    - jlpm playwright test
```

Comment:
Is there dependencies between the tests? Do we need to run the server tests first then frontend then integration ? 

If not it would better to split them up in 3 jobs , no? 
A) It would be easier to see what fails where. 
B) They would run faster because they run in parallel 


To reuse common setup across the 3 script use gitlab `extends` - https://docs.gitlab.com/ee/ci/yaml/?ref=jsramblings.com#extends

Reply:
Completely agree. 
Can you also confirm that all 3 tests suits output reports in junit format?

Reply:
Great suggestion! I split the tests into 3 separate stages (server tests, frontend tests and integration tests). Each of them now generates JUnit reports in the vdk-jupyter/vdk-jupyterlab-extension/ directory of the plugin. Tested it locally - it works.